include/openrave/openrave.h

```diff
@@ -1172,20 +1190,24 @@ class OPENRAVE_API IkParameterization
     }
     inline void SetDirection3D(const Vector& dir) {
         _type = IKP_Direction3D; _transform.rot = dir;
-    }
+        ENSURE_3DVEC_UNIT_LENGTH(_transform.rot);
+     }
     inline void SetRay4D(const RAY& ray) {
         _type = IKP_Ray4D; _transform.trans = ray.pos; _transform.rot = ray.dir;
-    }
+        ENSURE_3DVEC_UNIT_LENGTH(_transform.rot);
```

Comment:
@kanbouchou How about we put this check in the constructor of `class ray` too?

Reply:
Thanks, I added a check similar to quaternion class, but not exactly ``ENSURE_3DVEC_UNIT_LENGTH`` because ``ray`` class is templated and ``ENSURE_3DVEC_UNIT_LENGTH`` could not be applied.